src/nvim/spellfile.c

```diff
@@ -681,6 +681,7 @@ slang_T *spell_load_file(char *fname, char *lang, slang_T *old_lp, bool silent)
     res = 0;
     switch (n) {
     case SN_INFO:
+      XFREE_CLEAR(lp->sl_info);
```

Comment:
Using `XFREE_CLEAR` is a good idea because it reduces the risk of "double free" in these non-obvious `while` loops. 👍